internal/admission/validation/ingress/ingress.go

```diff
@@ -34,6 +43,10 @@ func ValidateIngress(
 			errMsgs = append(errMsgs, msg)
 		}
 	}
+	// Collect failures from the translation.
+	for _, failure := range failuresCollector.PopResourceFailures() {
+		errMsgs = append(errMsgs, failure.Message())
```

Comment:
If there are multiple Kong gateways, will this produce duplicate error messages?

Reply:
No, it's validated only once against a random client (it doesn't change in this PR).